user/languages/_includes/modern-c-cpp-toolchain.md

```diff
@@ -0,0 +1,130 @@
+### Modern C/C++ and Toolchain Versioning
+
+If you need a more modern toolchain or more granular control over your tools, you will have to handle that yourself. Fortunately, the community has already whitelisted the [repositories](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json) and [packages](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise) necessary for most such upgrades.
+
+#### GCC on Linux
+
+Ubuntu 12.04 shipped with GCC 4.6.3 and Ubuntu 14.04 shipped with GCC 4.8.2. If you need a more recent version (if you are working with C++11, you *must* do this; support for C++11 was not finished until GCC 4.9), you will want to follow these directions:
```

Comment:
A few comments:

1. I think we should use present tense here: Ubuntu 12.04 _ships_ with … While highly unlikely, these versions *could* change if Ubuntu team updates the packages.
1. While I agree with the sentiment, I don't think this parenthetical remark belongs here.
1. This YAML fragment is not *instructions* per se. And you don't have to have multiple jobs via `matrix.include` to realize C++-11 support, either. I agree with @plaindocs here; this is not a *must*.

I am guessing that, here, _this_ means "realizing the C++-11 support". I think this subject should be treated in a separate section.